crates/templates/src/manager.rs

```diff
@@ -487,6 +517,97 @@ mod tests {
         ));
     }
 
+    #[tokio::test]
+    async fn can_list_all_templates_with_empty_tags() {
+        let temp_dir = tempdir().unwrap();
+        let store = TemplateStore::new(temp_dir.path());
+        let manager = TemplateManager { store };
+        let source = TemplateSource::File(project_root());
+
+        manager
+            .install(&source, &InstallOptions::default(), &DiscardingReporter)
+            .await
+            .unwrap();
+
+        let list_results = manager.list_with_tags(&[]).await.unwrap();
+        assert_eq!(0, list_results.skipped.len());
+        assert_eq!(TPLS_IN_THIS, list_results.templates.len());
+    }
+
+    #[tokio::test]
+    async fn skips_when_all_tags_do_not_match() {
+        let temp_dir = tempdir().unwrap();
+        let store = TemplateStore::new(temp_dir.path());
+        let manager = TemplateManager { store };
+        let source = TemplateSource::File(project_root());
+
+        manager
+            .install(&source, &InstallOptions::default(), &DiscardingReporter)
+            .await
+            .unwrap();
+
+        let tags_to_match = vec!["c".to_string(), "".to_string()];
```

Comment:
I'd suggest something like `"unused_tag"` or `"nonexistent_tag_1234"` - to me that feels more explanatory than the empty string?